### PR TITLE
feat(entities): add hierarchy zones

### PR DESCRIPTION
**Pull request by caspiano**
- Adds placeholder `Model::Zone` for each hierarchical zone tag. i.e. `building`, `org`, `level`, `area`
- Removes `Test` prefix from development entities

### CODE REVIEW

**Review by dukeraphaelng — APPROVED** [review]

**Comment by kimburgess** [comment]
Is it worth popping in a `building` and `level` tagged zone here too to provide the full base hierarchy?

**Comment by caspiano** [comment]
Good idea @kimburgess

File: src/tasks/entities.cr [hunk 1]
```diff
@@ -191,17 +191,18 @@ module PlaceOS::Tasks::Entities
...
     end
 
-    zone = upsert_document(Model::Zone.all) do
-      zone_name = "TestZone-#{version}"
-      new_zone = Model::Zone.new(name: zone_name)
-      new_zone
+    # Create default entities for each element of the hierarchy
+    zones = {"building", "org", "level", "area"}.map do |tag|
+      upsert_document(Model::Zone.with_tag(tag)) do
+        Model::Zone.new.tap do |zone|
```
Comment (kimburgess) [inline src/tasks/entities.cr:197]: Should these be setting up the `parent` hierarchy too?

File: src/tasks/entities.cr [hunk 2]
```diff
@@ -213,38 +214,37 @@ module PlaceOS::Tasks::Entities
...
+        trigger_instance.trigger = trigger
+      end
     end
 
     upsert_document(Model::Edge.all) do
       Model::Edge.new(
-        name: "TestEdge-#{version}",
-        description: "Automatically generated edge profile. Set PLACE_EDGE_SECRET in your edge node's",
+        name: "Edge-#{version}",
+        description: "Automatically generated Edge profile. Set PLACE_EDGE_SECRET in your edge node's",
```
Comment (kimburgess) [inline src/tasks/entities.cr:247]:
```suggestion
        description: "Automatically generated Edge profile. Set PLACE_EDGE_SECRET on your edge nodes",
```

File: src/tasks/entities.cr [hunk 3]
```diff
@@ -149,12 +149,12 @@ module PlaceOS::Tasks::Entities
       name: private_repository_name,
       folder_name: private_repository_folder_name,
     )) do
-      repo = Model::Generator.repository(type: Model::Repository::Type::Driver)
-      repo.uri = private_repository_uri
-      repo.name = private_repository_name
-      repo.folder_name = private_repository_folder_name
-      repo.description = "PlaceOS Private Drivers"
-      repo
+      Model::Generator.repository(type: Model::Repository::Type::Driver).tap do |repo|
```
[the remark this answers is not in the document]
Comment (caspiano) [reply src/tasks/entities.cr:152]: I think `tap` makes sense here

File: src/tasks/entities.cr (hunk 1 above)
Comment (caspiano) [reply src/tasks/entities.cr:197]: Purpose of this is for configuring `source`, parent association isn’t strictly required for the development environment. These entities are ignored when the env is `production`

**Pull request merged**